hummingbot/market/ddex/ddex_market.pyx

```diff
@@ -240,7 +240,7 @@ cdef class DDEXMarket(MarketBase):
                 )
 
     def _update_balances(self):
-        self._account_balances = {k: Decimal(repr(v)) for k, v in self.wallet.get_all_balances().items()}
+        self._account_balances = {k: v for k, v in self.wallet.get_all_balances().items()}
```

Comment:
you can just use `self.wallet.get_all_balances().copy()` in this case. The contents are just the same... you just want to copy it once to isolate them.